### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Shevtsooov**
[DEMO LINK](https://Shevtsooov.github.io/react_todo-app-with-api/)

### CODE REVIEW

**Review by silchyn — CHANGES_REQUESTED** [review]
Why temporary todo appears on the bottom if you put the actual todo at the top?

**Comment by Shevtsooov** [comment]
@silchyn Oops, forgot to fix it. Thanks.

**Review by silchyn — CHANGES_REQUESTED** [review]
image

**Comment by Shevtsooov** [comment]
@silchyn Hi, I am sorry, but I am not quite sure where the issue is.
Could you elaborate it on, please?

**Comment by Shevtsooov** [comment]
@silchyn Hi, thank you, fixed the issue.

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,221 @@
...
 export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<TodoStatus>(TodoStatus.all);
+  const [errorMessage, setErrorMessage]
+  = useState<ErrorMessage>(ErrorMessage.noError);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletedTodoId, setDeletedTodoId] = useState<number[]>([]);
+  const [updatingTodoIds, setUpdatingTodoIds] = useState<number[]>([]);
```
Comment (dinaminahi) [inline src/App.tsx:33]: Can we use just one state for both deleting and updating?

File: src/components/Header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    <form
+      onSubmit={handleSubmit}
+    >
+      <input
+        type="text"
+        className="todoapp__new-todo"
+        placeholder="What needs to be done?"
+        disabled={isLoading}
+        value={newTodoTitle}
+        onChange={(e) => setNewTodoTitle(e.target.value)}
```
Comment (dinaminahi) [inline src/components/Header/Header.tsx:41]: Create  a handler for that

File: src/App.tsx [hunk 3]
```diff
@@ -1,24 +1,221 @@
...
 
-const USER_ID = 0;
+const USER_ID = 10890;
 
 export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<TodoStatus>(TodoStatus.all);
+  const [errorMessage, setErrorMessage]
+  = useState<ErrorMessage>(ErrorMessage.noError);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
```
Comment (dinaminahi) [inline src/App.tsx:29]: Let's move this to Header component

**Review by dinaminahi — CHANGES_REQUESTED** [review]
Good job 👍 
let's add few improvements

File: src/App.tsx [hunk 4]
```diff
@@ -1,24 +1,219 @@
...
+import { TodoStatus } from './types/TodoStatus';
 
-const USER_ID = 0;
+const USER_ID = 10890;
 
 export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<TodoStatus>(TodoStatus.all);
+  const [errorMessage, setErrorMessage]
+  = useState<ErrorMessage>(ErrorMessage.noError);
```
Comment (naziamoff) [inline src/App.tsx:27]:
```suggestion
  const [
    errorMessage, 
    setErrorMessage
  ] = useState<ErrorMessage>(ErrorMessage.noError);
```

Comment (naziamoff) [reply src/App.tsx:27]: nice thing that you created enum for error 🔥

File: src/App.tsx [hunk 5]
```diff
@@ -1,24 +1,219 @@
...
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [delUpdTodoIds, setDelUpdTodoIds] = useState<number[]>([]);
```
Comment (naziamoff) [inline src/App.tsx:30]: whts ths? dnt ndrstnd

File: src/App.tsx [hunk 6]
```diff
@@ -1,24 +1,219 @@
...
+
+  //   addTodos(USER_ID, {
+  //     title: newTodoTitle,
+  //     userId: USER_ID,
+  //     completed: false,
+  //   })
+  //     .then((result) => {
+  //       setVisibleTodos((prevTodos) => {
+  //         return [result, ...prevTodos];
+  //       });
```
Comment (naziamoff) [inline src/App.tsx:87]: You don't want to see commented code. It is confusing af, as you never now what the code owner wanted to tell with it

File: src/components/Error/Error.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    return () => {
+      clearTimeout(delay);
+    };
+  }, [handleCloseError]);
+
+  return (
+    <div
+      className={(cn(
+        'notification', 'is-danger', 'is-light', 'has-text-weight-normal', {
```
Comment (naziamoff) [inline src/components/Error/Error.tsx:27]:
```suggestion
        'notification is-danger is-light has-text-weight-normal', {
```

File: src/types/TodoStatus.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+export enum TodoStatus {
+  all = 'all',
+  active = 'active',
+  completed = 'completed',
+}
```
Comment (naziamoff) [inline src/types/TodoStatus.ts:5]: Todo names usually start from capital letter
```suggestion
export enum TodoStatus {
  All = 'all',
  Active = 'active',
  Completed = 'completed',
}
```

File: src/types/ErrorMessage.ts [hunk 9]
```diff
@@ -0,0 +1,8 @@
+export enum ErrorMessage {
+  noError = '',
+  loadError = 'Unable to load a todo',
+  addError = 'Unable to add a todo',
+  deleteError = 'Unable to delete a todo',
+  updateError = 'Unable to update a todo',
+  titleError = 'Title can\'t be empty',
```
Comment (naziamoff) [inline src/types/ErrorMessage.ts:7]: you should capitalize these

**Review by naziamoff — APPROVED** [review]
looks nice 🔥 
See some suggestions on how you can improve the code

**Pull request closed without merging**